app/decorators/prosecution_case_decorator.rb

```diff
@@ -23,4 +21,17 @@ def hearings_with_day_by_datetime
   def cracked?
     hearings.map { |h| h.cracked_ineffective_trial&.cracked? }.any?
   end
+
+  private
+
+  def sorter
```

Comment:
i was hoping this conditional would be got rid of by the refactor to use classes

Reply:
This is fundamentally a factory and should be a class method belonging to another class or module

Reply:
e.g.
```
TableSorter.for(sort_order)
```

Reply:
```
module TableSorters
  def self.for(field = nil)
    case field
    when /^type/
      HearingsTypeSorter.new(hearings, field)
    when /^provider/
      HearingsProviderSorter.new(hearings, field)
    else
      HearingsDateSorter.new(hearings, field)
    end
  end
end
```

Reply:
👍 As suggested, I have used the factory pattern to move the case statement to HearingsSorter